### ISSUE TITLE
create smartAPI record for idisk

### ISSUE THREAD

**Issue by andrewsu** [biothings/biothings_explorer#261]
Now that the new pending API for idisk is complete (https://github.com/biothings/pending.api/issues/22), creating an issue to track the creation of a smartAPI record and integration into BTE as a data source.

**Comment by colleenXu** [comment]
note that [infores IDs](https://docs.google.com/spreadsheets/d/1Ak1hRqlTLr1qa-7O0s5bqeTHukj9gSLQML1-lg6xIHM/edit#gid=293462374) are currently draft for this api and idisk as a knowledge source...

**Comment by colleenXu** [comment]
[EDITED] 

After testing, the [yaml](https://github.com/NCATS-Tangerine/translator-api-registry/blob/master/idisk/smartapi.yaml) works correctly. However, there is a bug with BTE...

It looks like when BTE cannot do ID resolution on the input ID, it'll incorrectly add the ID prefix to the sub-query for this API. 

---

**Incorrect prefix is being added to:**
- query starting from Disease MEDDRA:10017947 (BTE didn't give equivalent IDs...) to SmallMolecule
- this appears to be a particular issue with "has_adverse_effect_on-rev" operation, since the MEDDRA Disease IDs are very general and BTE is not finding equivalent IDs for them...

Correct (no prefix) happens for:
- query starting from Disease UMLS:C0016053 (Fibromylagias, BTE can ID resolve it to many IDs) to SmallMolecule. Correctly does the "is_effective_for-rev" operation
- query starting from Disease UMLS:C0027497 (Nausea, BTE can ID resolve it to an HP ID) to SmallMolecule. Correctly does the "has_adverse_reaction-rev" operation
- query starting from SmallMolecule UMLS:C0010137 (cortisone, BTE can ID resolve it to many IDs) to SmallMolecule. Correctly does the "interacts_with" and "interacts_with-rev" operations. 

Log showing incorrect sub-query done: 
the q should be just "10017947", not with the MEDDRA prefix...

```
        {
            "timestamp": "2021-08-23T04:43:31.523Z",
            "level": "DEBUG",
            "message": "call-apis: Succesfully made the following query: {\"url\":\"https://biothings.ncats.io/idisk/query\",\"params\":{\"fields\":\"name,umls\",\"size\":1000},\"data\":\"q=MEDDRA:10017947&scopes=has_adverse_effect_on.meddra\",\"method\":\"post\",\"timeout\":50000}",
            "code": null
        },
```

---

Setup: 
- convert this [yaml](https://github.com/NCATS-Tangerine/translator-api-registry/blob/master/idisk/smartapi.yaml) to json and save to /bte-trapi-workspace/packages/@biothings-explorer/bte-trapi/test/smartapi.json
- run a local instance of BTE and POST to the http://localhost:3000/test/query endpoint

Query:
```
{
    "message": {
        "query_graph": {
            "nodes": {
                "n0": {
                    "categories": ["biolink:Disease"],
                    "ids": ["MEDDRA:10017947"]
                },
                "n1": {
                    "categories": ["biolink:SmallMolecule"]
                }
            },
            "edges": {
                "e01": {
                    "subject": "n0",
                    "object": "n1"
                }
            }
        }
    }
}
```

**Comment by colleenXu** [comment]
Andrew and I discussed this today. We decided to register this [API](http://smart-api.info/registry?q=32f36164fabed5d3abe6c2fd899c9418), and integrate it with BTE here #262 

The bug mentioned above may be addressed by #249 and https://github.com/biothings/call-apis.js/issues/26. We can see if the bug continues to show up after and address it then...
